core/src/test/scala/kanaloa/reactive/dispatcher/IntegrationTests.scala

```diff
@@ -423,11 +425,14 @@ object IntegrationTests {
       expectMsg(shutdownTimeout, ShutdownSuccessfully)
     }
 
-    def performMultipleTests(test: ⇒ Int, targetSize: Int, numberOfTests: Int = 4) = {
+    def performMultipleTests(test: ⇒ Int, optimalSize: Int, numberOfTests: Int = 4) = {
+
+      val targetSize = optimalSize + 1 //target pool size is a bit larger to fully utilize the optimal concurrency
       val sizeResults = (1 to numberOfTests).map(_ ⇒ test)
 
       //normalized distance from the optimal size
       def distanceFromOptimalSize(actualSize: Int): Double = {
+        println("actual size:" + actualSize + "  optimal size: " + targetSize)
```

Comment:
I intentionally left this there. It gives more fine results than just the black and white test results, which is hard to control in a system that is designed to behave with randomness.